File: JMMinistry.Common/Dtos/DiscipleJourney/CycleSessionDto.cs
namespace JMMinistry.Common.Dtos.DiscipleJourney
{
    public class CycleSessionDto
    {
        public int Id { get; set; }
        public int StepCycleId { get; set; }
        public DateOnly Date { get; set; }
        public string? Topic { get; set; }
    }
}

File: jm_ministry_app/lib/features/training/domain/cycle_session_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'cycle_session_dto.freezed.dart';
part 'cycle_session_dto.g.dart';

@freezed
abstract class CycleSessionDto with _$CycleSessionDto {
  const factory CycleSessionDto({
    required int id,
    required int stepCycleId,
    required DateTime date,
    String? topic,
  }) = _CycleSessionDto;

  factory CycleSessionDto.fromJson(Map<String, dynamic> json) =>
      _$CycleSessionDtoFromJson(json);
}
